### PR TITLE
X672 sample collection date

### PR DESCRIPTION
**Pull request by seenanair**
Date of sample collection for fetal samples in block registration page
 -Not required for Slide registration
 -If fetal option is selected, then display UI field  to select date. Otherwise, it is not required
- Don't allow past dates for sample collection
- Don't allow user to register a fetal block without a sample collection date

### CODE REVIEW

File: src/pages/registration/RegistrationForm.tsx [hunk 1]
```diff
@@ -56,7 +56,7 @@ const RegistrationForm = ({ registrationInfo }: RegistrationFormParams) => {
   const isHMDMCEnabled = values.tissues[currentIndex].species === "Human";
   useEffect(() => {
     if (!isHMDMCEnabled) {
-      setFieldValue(`tissues.${currentIndex}.hmdmc`, "", true);
+      setFieldValue(`tissues.${currentIndex}.hmdmc`, "", false);
```
Comment (khelwood) [inline src/pages/registration/RegistrationForm.tsx:59]: Why this change?

Comment (seenanair) [reply src/pages/registration/RegistrationForm.tsx:59]: This is not required. I am reverting it back. Thank you for the finding. 👍

File: cypress/integration/shared/registration.ts [hunk 2]
```diff
@@ -41,6 +43,32 @@ export function shouldBehaveLikeARegistrationForm() {
...
+              isSlideRegistration ? "not.exist" : "be.visible"
+            );
+          }
+        );
+      });
+      context("when Adult is selected", () => {
+        before(() => {
+          cy.findByLabelText("Adult").click();
+        });
+        it("shows sample collection date", () => {
```
Comment (khelwood) [inline cypress/integration/shared/registration.ts:66]: "does not show sample collection date"

Comment (seenanair) [reply cypress/integration/shared/registration.ts]: Done. Thank you 👍

**Review by khelwood — APPROVED** [review]
